### DISCUSSION TITLE
working Hardware Seeed Studio XIAO ESP32S3 Sense ?

### DISCUSSION THREAD

**Discussion by 0VV3NZ** [jomjol/AI-on-the-edge-device#2476] [Q&A]
Hi, auf der suche nach ein ESP32 CAM habe ich dieses Modul entdeckt. 
Ich wollte gerne von den Profis wissen ob diese Hardware kompatiebel ist.

[https://www.seeedstudio.com/XIAO-ESP32S3-Sense-p-5639.html](url)

Hi, looking for a ESP32 CAM I discovered this module. 
I would like to know from the pros if this hardware is compatible.

**Comment by caco3** [comment]
Ich habe kein ESP32S3 und kann daher nicht sagen, ob es klappt oder nicht.
Die Kamera `OV2640` ist grundsätzlich supported.

Und wir haben keinen Code drin, der andere Module explizit verweigern würde.

**Comment by dadcoachengineer** [comment]
Trying to flash this module with errors.

➜  AI-on-the-edge-device__manual-setup__v15.7.0 esptool.py --chip esp32s3 --port /dev/cu.usbmodem11201 write_flash 0x01000 bootloader.bin 0x08000 partitions.bin 0x10000 firmware.bin
esptool.py v4.7.0
Serial port /dev/cu.usbmodem11201
Connecting...
Chip is ESP32-S3 (QFN56) (revision v0.2)
Features: WiFi, BLE, Embedded PSRAM 8MB (AP_3v3)
Crystal is 40MHz
MAC: dc:da:0c:13:d7:8c
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Unexpected chip id in image. Expected 9 but value was 0. Is this image for a different chip model?

A fatal error occurred: bootloader.bin is not an ESP32-S3 image. Use --force to flash anyway.

**Reply by caco3** [reply]
I am sure you would have to compile it for the ESP32S3 to have at least a change to have it working.

**Comment by SybexX** [comment]
https://github.com/Dag0d/AI-on-the-edge-device-ESP32s3/tree/rolling

**Reply by caco3** [reply]
@Dag0d If your modification works and only requires a different build configuration, then we could add it to our pipeline and also let it automatically be built.

**Reply by larryson309** [reply]
I tried this, but I get to many errors in platform.IO. I'm at the beginning of any programmer-skills and some of the instructions confused me. So I tried it further. 
Thx all of you for your work and time!

**Reply by SybexX** [reply]
https://github.com/Dag0d/AI-on-the-edge-device-ESP32s3/actions/runs/7960059040

**Reply by larryson309** [reply]
Hi, are there any Updates? @SybexX @Dag0d

**Reply by Dag0d** [reply]
I had it running on the Xiao ESP32S3 Sense board but the result was not great, the little board is running way too hot, which even causes crashes from time to time depending on the load and the duration.
my main reason behind that board was the integrated battery controller, which was also a disappointment, as it consumes way too much power. We would have to integrate proper deep sleep and everything.

I'm currently working on a different approach for the battery problem, I designed a ESP32 Cam "Hat" which integrates proper power management with a Mosfet which controls the power to the ESP32Cam.
I´m currently testing this and improving the firmware running on it´s microcontroller.
As soon as all issues are ironed out I will release the schematics and the software.

**Reply by wiktor-m** [reply]
Could you share your experience or schematics, etc.? I also want to build one using the "Xiao ESP32-S3 Sense" with solar power, as the gas meter is outside and doesn't have mains power.

**Comment by larryson309** [comment]
Hi, ich versuche ebenfalls, AI-on-the-Edge auf dem Xiao ESP32S3 zum laufen zu bringen, leider ohne Erfolg. 
Würde mich freuen, wenn es dann funktioniert. 

Hi, I am also trying to get AI-on-the-Edge to run on the Xiao ESP32S3, unfortunately without success. 
I would be happy if it works then. 
Thanks for you replies.

**Comment by SybexX** [comment]
momentan leider keine Zeit um daran weiter zu machen, versuche mal die firmware von @Slider0007 :
https://github.com/Slider0007/AI-on-the-edge-device/tree/esp32s3wroom
https://github.com/Slider0007/AI-on-the-edge-device/actions/runs/10407913299

**Reply by JH4711** [reply]
Hi guys, I was also keen on getting the AI-on-the-edge running on the Xiao ESP32S3-Sense Board, because it is a nifty tiny board, which has already SD-card reader and USB-C integrated. Since I have never used platformio before, I had a steep learning curve the past few days to modify the code accordingly. Since I am also not very familiar to github, I don't know, how to contribute and upload the modification I did. If you guide me, I could add my modifications to this interesting project! Main modifications: Define the env:seeed_xiao_esp32s3 in platformio.ini with many specific entries for the ESP32S3. Added a board-definition BOARD_XIAO_ESP32S3_SENSE with the appropriate IO-definition. Modified the SD-access in main.cpp, because SPI is used for the SD-card. And in Helper.cpp (and the corresponding CMakeLists.txt) I applied the modifications to support the on-die temperature sensor. For the hardware I added a white LED (2200 mCd) on GPIO_2 in series with a 51 Ohm resistor to GND. It is now basically up and running. Communication via WiFi and MQTT is working. BR, JH4711

**Reply by wiktor-m** [reply]
Can you share your code on github?

**Reply by JH4711** [reply]
Although I am willing to share, I am not familiar to github, how to share it... However, in the meantime I noticed that there is already a fork, which supports the ESP32S3: https://github.com/jomjol/AI-on-the-edge-device/discussions/3874.
But I haven't checked it, yet.
